feat: support new rangeStrategy=in-range-only (#13257)
renovatebot/renovate: docs/usage/configuration-options.md

- `replace` = Replace the range with a newer one if the new version falls outside it, and update nothing otherwise
- `widen` = Widen the range with newer one, e.g. `^1.0.0` -> `^1.0.0 || ^2.0.0`
- `update-lockfile` = Update the lock file when in-range updates are available, otherwise `replace` for updates out of range. Works for `bundler`, `composer`, `npm`, `yarn`, `terraform` and `poetry` so far

> - `in-range-only` = Update the lock file when in-range updates are available, ignore package file updates.

Reviewer: ```suggestion
- `in-range-only` = Update the lock file when in-range updates are available, ignore package file updates
```

We don't use a full stop at the end of list items by the way. 😉

This feature supports simple caret (`^`) and tilde (`~`) ranges only, like `^1.0.0` and `~1.0.0`.

> We discourage using of the `in-range-only` strategy unless you strictly need it.

Reviewer: How would a user "know" if they really need the `in-range-only` strategy?

The current docs don't really explain if/when/why you would need this strategy.

Author: Hi @HonkingGoose I think I tried to explain how this strategy works in this file: docs/usage/configuration-options.md:

- `in-range-only` = Update the lock file when in-range updates are available, ignore package file updates.

Reviewer: That is good, but it still doesn't explain why a user of Renovate would want to use the `in-range-only` option.

We as Renovate maintainers/programmers need to explain to our users when they "strictly need" this option. 😉 I think it's too easy on our part to just say: "figure out yourself if you need this or not". 😄

Author: Thanks @HonkingGoose I've tried to expand the explanation of the usage of the `in-range-strategy` in docs/usage/configuration-options.md

This feature supports simple caret (`^`) and tilde (`~`) ranges only, like `^1.0.0` and `~1.0.0`.

> The `in-range-only` strategy may be useful for people preferring to leave the package file unchanged and only do `update-lockfile` within the existing range.

Reviewer: ```suggestion
The `in-range-only` strategy may be useful if you want to leave the package file unchanged and only do `update-lockfile` within the existing range.
```

The `in-range-only` strategy behaves like `update-lockfile`, but discards any updates where the new version of the dependency is not equal to the current version.

> Please note that we discourage using of the `in-range-only` strategy unless you strictly need it.

Reviewer: ```suggestion
We recommend you avoid using the `in-range-only` strategy unless you strictly need it.
```

> Using the `in-range-only` strategy may result in being multiple releases behind without knowing it.

Reviewer: ```suggestion
Using the `in-range-only` strategy may result in you being multiple releases behind without knowing it.
```
